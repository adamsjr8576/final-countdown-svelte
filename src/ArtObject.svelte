<script>
  export let description;
  export let colors = [];
  export let period;
  export let image = '';
  export let title;
  export let century;
  export let medium;
</script>

<article>
  <img src={image} alt={title} />
    <div class='colors-container'>
      {#each colors as color}
        <div class="color-div" style="background-color: {color.color}"></div>
      {/each}
    </div>
    <div>
      <h4>Title:</h4>
      <p>{title}</p>
    </div>
    <div>
      <h4>Time:</h4>
      <p>{century} - {period}</p>
    </div>
    <div>
      <h4>Medium: </h4>
      <p>{medium}</p>
    </div>
    <div>
      <h4>Description: </h4>
      <p class="description-p">{description}</p>
    </div>
</article>

<style>

  article {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 350px;
    height: 550px;
    margin: 15px;
    border: 2px solid black;
    background-color: #F6F6FB;
    overflow: auto;
  }

  article:hover {
    border: 2px solid #ff3e00;
  }

  img {
    height: auto;
    width: 350px;
  }

  .colors-container {
    width: 95%;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .color-div {
    height: 25px;
    width: 25px;
    margin: 0;
  }

  h4 {
    text-decoration: underline;
    margin: 5px 0;
  }

  p {
    margin: 0;
  }

  .description-p {
    height: 100px;
    overflow: auto;
    width: 90%;
    border: 1px solid black;
    margin: auto;
    padding: 10px
  }

  div {
    width: 90%;
    margin: 10px 0;
  }

</style>
