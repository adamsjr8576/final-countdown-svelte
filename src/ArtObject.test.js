import { render, fireEvent } from '@testing-library/svelte';
import ArtObject from './ArtObject.svelte';

describe('App', () => {
  it('should match the snapshot without colors or image passed as props', () => {
    const wrapper = render(ArtObject, {
      description: 'A woman with a creepy smile',
      period: 'Romantic Period',
      title: 'The Mona Lisa',
      century: '17th CE',
      medium: 'Paint on Canvas'
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with colors or image passed as props', () => {
    const wrapper = render(ArtObject, {
      description: 'A woman with a creepy smile',
      colors: ['#93453', '#34523'],
      period: 'Romantic Period',
      image: 'monalisa.jpg',
      title: 'The Mona Lisa',
      century: '17th CE',
      medium: 'Paint on Canvas'
    });

    expect(wrapper).toMatchSnapshot();
  });

});
