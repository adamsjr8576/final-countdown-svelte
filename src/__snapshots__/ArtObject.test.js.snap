// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with colors or image passed as props 1`] = `
Object {
  "component": ArtObject {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        "A woman with a creepy smile",
        Array [
          "#93453",
          "#34523",
        ],
        "Romantic Period",
        "monalisa.jpg",
        "The Mona Lisa",
        "17th CE",
        "Paint on Canvas",
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_mount": Array [],
      "props": Object {
        "century": 5,
        "colors": 1,
        "description": 0,
        "image": 3,
        "medium": 6,
        "period": 2,
        "title": 4,
      },
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
    "$set": [Function],
  },
  "container": <body>
    <div>
      <article
        class="svelte-121edkc"
      >
        <img
          alt="The Mona Lisa"
          class="svelte-121edkc"
          src="monalisa.jpg"
        />
         
        <div
          class="colors-container svelte-121edkc"
        >
          <div
            class="color-div svelte-121edkc"
          />
          <div
            class="color-div svelte-121edkc"
          />
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Title:
          </h4>
           
          <p
            class="svelte-121edkc"
          >
            The Mona Lisa
          </p>
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Time:
          </h4>
           
          <p
            class="svelte-121edkc"
          >
            17th CE
             - 
            Romantic Period
          </p>
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Medium:
          </h4>
           
          <p
            class="svelte-121edkc"
          >
            Paint on Canvas
          </p>
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Description:
          </h4>
           
          <p
            class="description-p svelte-121edkc"
          >
            A woman with a creepy smile
          </p>
        </div>
      </article>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should match the snapshot without colors or image passed as props 1`] = `
Object {
  "component": ArtObject {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        "A woman with a creepy smile",
        Array [],
        "Romantic Period",
        "",
        "The Mona Lisa",
        "17th CE",
        "Paint on Canvas",
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_mount": Array [],
      "props": Object {
        "century": 5,
        "colors": 1,
        "description": 0,
        "image": 3,
        "medium": 6,
        "period": 2,
        "title": 4,
      },
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
    "$set": [Function],
  },
  "container": <body>
    <div>
      <article
        class="svelte-121edkc"
      >
        <img
          alt="The Mona Lisa"
          class="svelte-121edkc"
        />
         
        <div
          class="colors-container svelte-121edkc"
        />
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Title:
          </h4>
           
          <p
            class="svelte-121edkc"
          >
            The Mona Lisa
          </p>
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Time:
          </h4>
           
          <p
            class="svelte-121edkc"
          >
            17th CE
             - 
            Romantic Period
          </p>
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Medium:
          </h4>
           
          <p
            class="svelte-121edkc"
          >
            Paint on Canvas
          </p>
        </div>
         
        <div
          class="svelte-121edkc"
        >
          <h4
            class="svelte-121edkc"
          >
            Description:
          </h4>
           
          <p
            class="description-p svelte-121edkc"
          >
            A woman with a creepy smile
          </p>
        </div>
      </article>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
